core/templates/dev/head/pages/email-dashboard-pages/email-dashboard-data.service.ts

```diff
@@ -16,109 +16,127 @@
-        return currentPageIndex;
-      },
-
-      getLatestCursor: function() {
-        return latestCursor;
-      },
-
-      submitQuery: function(data) {
-        var startQueryIndex = currentPageIndex * QUERIES_PER_PAGE;
-        var endQueryIndex = (currentPageIndex + 1) * QUERIES_PER_PAGE;
-
-        return $http.post(QUERY_DATA_URL, {
-          data: data
-        }).then(function(response) {
-          var data = response.data;
-          var newQueries = [data.query];
-          queries = newQueries.concat(queries);
-          return queries.slice(startQueryIndex, endQueryIndex);
-        });
-      },
-
-      getNextQueries: function() {
-        var startQueryIndex = (currentPageIndex + 1) * QUERIES_PER_PAGE;
-        var endQueryIndex = (currentPageIndex + 2) * QUERIES_PER_PAGE;
-
-        if (queries.length >= endQueryIndex ||
-            (latestCursor === null && currentPageIndex !== -1)) {
-          currentPageIndex = currentPageIndex + 1;
-          return $q(function(resolver) {
-            resolver(queries.slice(startQueryIndex, endQueryIndex));
-          });
-        } else {
-          currentPageIndex = currentPageIndex + 1;
-          return fetchQueriesPage(QUERIES_PER_PAGE, latestCursor)
-            .then(function(data) {
-              queries = queries.concat(data.recent_queries);
-              latestCursor = data.cursor;
-              return queries.slice(startQueryIndex, endQueryIndex);
-            });
-        }
-      },
-
-      getPreviousQueries: function() {
-        var startQueryIndex = (currentPageIndex - 1) * QUERIES_PER_PAGE;
-        var endQueryIndex = currentPageIndex * QUERIES_PER_PAGE;
-        currentPageIndex = currentPageIndex - 1;
-        return queries.slice(startQueryIndex, endQueryIndex);
-      },
-
-      isNextPageAvailable: function() {
-        var nextQueryIndex = (currentPageIndex + 1) * QUERIES_PER_PAGE;
-        return (queries.length > nextQueryIndex) || Boolean(latestCursor);
-      },
-
-      isPreviousPageAvailable: function() {
-        return (currentPageIndex > 0);
-      },
-
-      fetchQuery: function(queryId) {
-        return $http.get(QUERY_STATUS_CHECK_URL, {
-          params: {
-            query_id: queryId
-          }
-        }).then(function(response) {
-          var data = response.data;
-          queries.forEach(function(query, index, queries) {
-            if (query.id === queryId) {
-              queries[index] = data.query;
-            }
-          });
-          return data.query;
+  getQueries(): Array<Query> {
+    return this.queries;
+  }
+
+  getCurrentPageIndex(): number {
+    return this.currentPageIndex;
+  }
+
+  getLatestCursor(): string {
+    return this.latestCursor;
+  }
+
+  // TODO(#7176): Replace 'any' with the exact type. This has been kept as
+  // 'any' because 'data' is a dict with underscore_cased
+  // keys which give tslint errors against underscore_casing in favor of
+  // camelCasing.
```

Comment:
Shouldn't this be a domain object which takes the backend dict (which would be underscore cased) and then store it as camelCased values?

Reply:
@nithusha21 I think we use domain objects to convert the response of `get` endpoints to objects but this is the data we have to send the backend and that data should be a dict not a object? L80

Reply:
Ah I see, I think we use dicts in other places too, where we send data to the backend. I think this is fine.